### PR TITLE
Improve docstring and validation of parameters in CameraGeometry

### PR DESCRIPTION
**Pull request by maxnoe**

### CODE REVIEW

**Review by kosack — APPROVED** [review]

File: ctapipe/instrument/camera/geometry.py [hunk 1]
```diff
@@ -88,25 +88,32 @@ class CameraGeometry:
...
+    pix_id : np.ndarray[int]
         pixels id numbers
-    pix_x : array with units
+    pix_x : u.Quantity
         position of each pixel (x-coordinate)
-    pix_y : array with units
+    pix_y : u.Quantity
         position of each pixel (y-coordinate)
-    pix_area : array(float)
-        surface area of each pixel, if None will be calculated
```
Comment (Tobychev) [inline ctapipe/instrument/camera/geometry.py:102]: was this "if not given will be calculated" simply not correct?

Comment (maxnoe) [reply ctapipe/instrument/camera/geometry.py:102]: Yes, this got removed a long time ago and made a required argument

File: ctapipe/instrument/camera/geometry.py [hunk 2]
```diff
@@ -120,43 +127,60 @@ def __init__(
...
+            if pix_area.shape != shape:
+                raise ValueError(
+                    f"pix_area has wrong shape: {pix_area.shape}, expected {shape}"
+                )
+
+            if isinstance(pix_type, str):
+                pix_type = PixelShape.from_string(pix_type)
+            elif not isinstance(pix_type, PixelShape):
+                raise TypeError(
+                    f"pix_type most be a PixelShape or the name of a PixelShape, got {pix_type}"
```
Comment (LukasNickel) [inline ctapipe/instrument/camera/geometry.py:169]: typo: ~most~ must

File: ctapipe/instrument/camera/geometry.py [hunk 3]
```diff
@@ -88,25 +88,32 @@ class CameraGeometry:
...
-    cam_rotation : overall camera rotation with units
+    pix_rotation : u.Quantity[angle]
+        rotation of the pixels, global value for all pixels
+    cam_rotation : u.Quantity[angle]
+        rotation of the camera. All coordinates will be interpreted
+        to be rotated by this angle with respect to the definition of the
+        frame the pixel coordinates are defined in.
+    neighbors : list[list[int]]
+        adjacency list for each pixel
+        If not given, will be build from the pixel width and distances
```
Comment (LukasNickel) [inline ctapipe/instrument/camera/geometry.py:111]: Is this correct? The init skips the neighbor-related stuff if the value is None, but there is code to calculate it, that gets called in the properties. I assume the code in the init only makes sure that any precomputed arrays are proper boolean sparse matrices?

**Review by LukasNickel — APPROVED** [review]

**Review by kosack — APPROVED** [review]

**Pull request merged**